### PR TITLE
Implement ability to update user principals

### PR DESCRIPTION
**Pull request by hpeebles**

### CODE REVIEW

File: backend/canisters/local_user_index/impl/src/lib.rs [hunk 1]
```diff
@@ -215,6 +218,10 @@ struct Data {
     pub rng_seed: [u8; 32],
 }
 
+fn identity_canister_id() -> CanisterId {
+    CanisterId::from_text("6klfq-niaaa-aaaar-qadbq-cai").unwrap()
```
Comment (megrogan) [inline backend/canisters/local_user_index/impl/src/lib.rs:222]: Use test flag to set this to prod_test id

Comment (hpeebles) [reply backend/canisters/local_user_index/impl/src/lib.rs:222]: The test flag isn't easily available at this stage.
I can just upgrade it first on prod test with the value set to the prod test canisterId.

File: backend/canisters/user_index/impl/src/lib.rs [hunk 2]
```diff
@@ -231,6 +240,10 @@ struct Data {
     pub diamond_membership_fees: DiamondMembershipFees,
 }
 
+fn identity_canister_id() -> CanisterId {
+    CanisterId::from_text("6klfq-niaaa-aaaar-qadbq-cai").unwrap()
```
Comment (megrogan) [inline backend/canisters/user_index/impl/src/lib.rs:244]: Init prod_test too

File: backend/canisters/user_index/impl/src/updates/c2c_update_user_principal.rs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    new_principal: Principal,
+}
+
+fn prepare(args: Args, state: &RuntimeState) -> PrepareResult {
+    let user_id = state.data.users.get_by_principal(&args.old_principal).unwrap().user_id;
+
+    PrepareResult {
+        user_id,
+        old_principal: args.old_principal,
+        new_principal: args.new_principal,
```
Comment (megrogan) [inline backend/canisters/user_index/impl/src/updates/c2c_update_user_principal.rs:46]: Seems a bit weird to echo back the args

**Review by megrogan — APPROVED** [review]

**Review by megrogan — APPROVED** [review]

**Pull request merged**